analysis/analysis.py

```diff
@@ -43,6 +43,77 @@ def compute_total_view_time(student_data):
     return total_view_time
 
 
+def get_word_frequency_differences(student_data):
+    """
+    Looks over the data and compares responses from people who have read the text vs.
+    people who have not read the text before this exercise
+    :return: a list of word frequency differences, by increasing order of frequency differences
+    """
+
+    # Iterate through all data, and separate ids of students who have vs. have not read the text
+    yes_id = []
+    no_id = []
+
+    for response in student_data:
+        if (response['question'].find('Have you encountered this text before') == 0
+                and response['context'].find('This is an ad.') == 0):
+            if response['response'].lower().find('yes') == -1:
+                no_id.append(response['student_id'])
+            else:
+                yes_id.append(response['student_id'])
+
+    # Iterate through all responses, store in list words used to describe the text for students who
+    # have vs. have not read the text
+    ad_yes_words = []
+    ad_no_words = []
+
+    for element in student_data:
```

Comment:
this loop and the above loop are going through the same data, so you could compress this into a single for loop for efficiency